File: Notes/CTCI/03-stacks-queues/queue.js
module.exports = function Queue() {
  const items = [];

  this.add = function(data) {
    items.push(data);
  }

  this.remove = function() {
    return items.shift();
  }

  this.peek = function() {
    return items[0];
  }

  this.isEmpty = function() {
    return items.length === 0;
  }
}

File: Notes/CTCI/03-stacks-queues/py/Queue.py
class Queue:
    def __init__(self):
        self.items = []

    def add(self, i):
        return self.items.append(i)

    def remove(self):
        return None if self.is_empty() else self.items.pop(0)

    def is_empty(self):
        return len(self.items) == 0

    def peek(self):
        return self.items[0]
